### PR TITLE
Fix `Error! Cannot convert undefined or null to object`

### PR DESCRIPTION
**Pull request by denyshld**
Fix `Error! Cannot convert undefined or null to object` error on no attachments object in message

### CODE REVIEW

File: lib/actions/streamToFile.js [hunk 1]
```diff
@@ -10,6 +10,13 @@ exports.process = async function (msg, cfg) {
   const client = new Client(this.logger, cfg);
   const bucketName = msg.body.bucketName ? msg.body.bucketName : cfg.bucketName;
 
+  if (!msg.attachments) {
```
Comment (uaArsen) [inline lib/actions/streamToFile.js:13]:
```suggestion
  if (!msg.attachments || Object.keys(msg.attachments).length === 0) {
```

File: lib/actions/streamToFile.js [hunk 2]
```diff
@@ -10,6 +10,13 @@ exports.process = async function (msg, cfg) {
...
+    throw new Error('Message has no attachments object (no attachments)');
```
Comment (uaArsen) [inline lib/actions/streamToFile.js:14]:
```suggestion
    throw new Error('Has no attachment');
```

File: lib/actions/streamToFile.js [hunk 3]
```diff
@@ -10,6 +10,13 @@ exports.process = async function (msg, cfg) {
...
+  } else if (Object.keys(msg.attachments).length > 1) {
+    throw new Error('More than 1 attachment');
+  } else if (Object.keys(msg.attachments).length === 0) {
```
Comment (uaArsen) [inline lib/actions/streamToFile.js:17]:
```suggestion
  }
```

File: lib/actions/streamToFile.js [hunk 4]
```diff
@@ -10,6 +10,13 @@ exports.process = async function (msg, cfg) {
...
+    throw new Error('Has no attachment');
```
Comment (uaArsen) [inline lib/actions/streamToFile.js:18]:
```suggestion
```

File: lib/actions/streamToFile.js [hunk 5]
```diff
@@ -10,6 +10,13 @@ exports.process = async function (msg, cfg) {
...
+  }
```
Comment (uaArsen) [inline lib/actions/streamToFile.js:19]:
```suggestion
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,9 @@
+# 1.2.3 (December 26, 2019)
+    * Fix `Error! Cannot convert undefined or null to object` error on no attachments object in message in Write file action
+
+## General Changes
+    * Update sailor version to 2.5.4
```
Comment (uaArsen) [inline CHANGELOG.md]: Cant see this changes in package.json. Also why not upgrading to sailor 2.6.3?

File: lib/actions/streamToFile.js (hunk 1 above)
Comment (denyshld) [reply lib/actions/streamToFile.js:13]: Pretty different error reasons

File: CHANGELOG.md (hunk 6 above)
Comment (denyshld) [reply CHANGELOG.md]: Stupid copypaste mistake. Upgrading is out of scope

**Review by shulkaolka — APPROVED** [review]

**Review by uaArsen — APPROVED** [review]

**Pull request merged**